common/lib/capa/capa/capa_problem.py

```diff
@@ -488,67 +500,69 @@ def do_targeted_feedback(self, tree):
-                if choice.get('name') == student_answer:
-                    expl_id_for_student_answer = choice.get('explanation-id')
-                if choice.get('correct') == 'true':
-                    solution_id = choice.get('explanation-id')
-
-            # Filter out targetedfeedback that doesn't correspond to the answer the student selected
-            # Note: following-sibling will grab all following siblings, so we just want the first in the list
-            targetedfeedbackset = mult_choice_response.xpath('./following-sibling::targetedfeedbackset')
-            if len(targetedfeedbackset) != 0:
-                targetedfeedbackset = targetedfeedbackset[0]
-                targetedfeedbacks = targetedfeedbackset.xpath('./targetedfeedback')
-                for targetedfeedback in targetedfeedbacks:
-                    # Don't show targeted feedback if the student hasn't answer the problem
-                    # or if the target feedback doesn't match the student's (incorrect) answer
-                    if not self.done or targetedfeedback.get('explanation-id') != expl_id_for_student_answer:
-                        targetedfeedbackset.remove(targetedfeedback)
-
-            # Do not displace the solution under these circumstances
-            if not show_explanation or not self.done:
-                continue
-
-            # The next element should either be <solution> or <solutionset>
-            next_element = targetedfeedbackset.getnext()
-            parent_element = tree
-            solution_element = None
-            if next_element is not None and next_element.tag == 'solution':
-                solution_element = next_element
-            elif next_element is not None and next_element.tag == 'solutionset':
-                solutions = next_element.xpath('./solution')
-                for solution in solutions:
-                    if solution.get('explanation-id') == solution_id:
-                        parent_element = next_element
-                        solution_element = solution
-
-            # If could not find the solution element, then skip the remaining steps below
-            if solution_element is None:
-                continue
-
-            # Change our correct-choice explanation from a "solution explanation" to within
-            # the set of targeted feedback, which means the explanation will render on the page
-            # without the student clicking "Show Answer" or seeing a checkmark next to the correct choice
-            parent_element.remove(solution_element)
-
-            # Add our solution instead to the targetedfeedbackset and change its tag name
-            solution_element.tag = 'targetedfeedback'
-            targetedfeedbackset.append(solution_element)
+        questions = tree.xpath('//problem/question')
+        for question in questions:
+            for mult_choice_response in question.xpath('//multiplechoiceresponse[@targeted-feedback]'):
```

Comment:
@cpennington as per understanding this is the right place for this code. Do you any issues?